code/game/machinery/kitchen/cooking_machines/_appliance.dm

```diff
@@ -0,0 +1,655 @@
+		burn_food(CI)
+
+	// Gotta hurt.
+	for(var/obj/item/holder/H in CI.container.contents)
+		var/mob/living/M = H.contained
+		if (M)
+			M.apply_damage(rand(1,3), mobdamagetype, "chest")
+
+	return 1
+
+/obj/machinery/appliance/process()
+	if (cooking_power > 0 && cooking)
+		for (var/i in cooking_objs)
+			do_cooking_tick(i)
+
+
+
+/obj/machinery/appliance/proc/finish_cooking(datum/cooking_item/CI)
+
+	src.visible_message("<span class='notice'>\The [src] pings!</span>")
+	if(cooked_sound)
+		playsound(get_turf(src), cooked_sound, 50, 1)
+	//Check recipes first, a valid recipe overrides other options
+	var/datum/recipe/recipe = null
+	var/atom/C = null
+	if (CI.container)
+		C = CI.container
+	else
+		C = src
+	recipe = select_recipe(available_recipes,C)
+
+	if (recipe)
+		CI.result_type = 4//Recipe type, a specific recipe will transform the ingredients into a new food
+		var/list/results = recipe.make_food(C)
+
+		var/atom/temp = new /atom //To prevent infinite loops, all results will be moved into a temporary location so they're not considered as inputs for other recipes
+
+		for (var/atom/movable/AM in results)
+			AM.loc = temp
+
+		//making multiple copies of a recipe from one container. For example, tons of fries
+		while (select_recipe(available_recipes,C) == recipe)
+			var/list/TR = recipe.make_food(C)
+			for (var/atom/movable/AM in TR) //Move results to buffer
+				AM.loc = temp
+			results.Add(TR)
+
+
+		for (var/r in results)
+			var/obj/item/reagent_containers/food/snacks/R = r
+			R.loc = C //Move everything from the buffer back to the container
+			qdel(temp) //delete buffer object
+			temp = null
+			R.cooked |= cook_type.
+
+		.=1 //None of the rest of this function is relevant for recipe cooking
+
+	else if(CI.combine_target)
+		CI.result_type = 3//Combination type. We're making something out of our ingredients
+		.=combination_cook(CI)
+
+
+	else
+		//Otherwise, we're just doing standard modification cooking. change a color + name
+		for (var/obj/item/i in CI.container)
+			modify_cook(i, CI)
+
+	//Final step. Cook function just cooks batter for now.
+	for (var/obj/item/reagent_containers/food/snacks/S in CI.container)
+		S.cook()
+
+
+//Combination cooking involves combining the names and reagents of ingredients into a predefined output object
+//The ingredients represent flavours or fillings. EG: donut pizza, cheese bread
+/obj/machinery/appliance/proc/combination_cook(datum/cooking_item/CI)
+	var/cook_path = output_options[CI.combine_target]
+
+	var/list/words = list()
+	var/list/cooktypes = list()
+	var/datum/reagents/buffer = new /datum/reagents(1000)
+	var/totalcolour
+
+	for (var/obj/item/I in CI.container)
+		var/obj/item/reagent_containers/food/snacks/S
+		if (istype(I, /obj/item/holder))
+			S = create_mob_food(I, CI)
+		else if (istype(I, /obj/item/reagent_containers/food/snacks))
+			S = I
+
+		if (!S)
+			continue
+
+		words |= dd_text2List(S.name," ")
```

Comment:
this is really stupid and this proc is useless and bad and slow get rid of it do this another way
also... https://www.byond.com/docs/ref/#/proc/splittext ????